File: learning-factory/warehouse/artifacts/projects/build-your-own-x/programming-language/mal-make-a-lisp-5bbc84338b/job_byox_build_v1_6cb4088c2ac9e03a348affc9f115897e/attempt-001/starter/sprig/vm.py
"""Stack virtual machine (milestone 4)."""


class VirtualMachine(object):
    def __init__(self, max_steps=10000):
        self.max_steps = max_steps

    def run(self, bytecode, env=None):
        raise NotImplementedError("milestone 4: VirtualMachine.run")

File: learning-factory/warehouse/artifacts/projects/build-your-own-x/remediation/54fbf20b587133619a1f/repair-v1/job_byox_repair_v1_g1_250e5b6563a90a708220b0592e8a0dfe/attempt-001/starter/src/vm.js
import { PebbleNotImplementedError } from "./errors.js";

export class VirtualMachine {
  constructor(options = {}) {
    this.options = options;
  }

  execute(bytecode) {
    // TODO: validate/execute the documented bytecode envelope and OpCode set,
    // collect emitted values, reject non-finite numeric results, enforce runtime
    // errors, and charge maxSteps once per dispatched instruction.
    void bytecode;
    throw new PebbleNotImplementedError("virtual machine");
  }
}
